bridges/evm/test/ObiLib.test.js

```diff
@@ -0,0 +1,198 @@
+        });
+
+        it("should decodeU128 correctly", async () => {
+            result = await this.forTest.decodeU128([0, "0x01000000000000000000000000000000"]);
+            result.toString().should.eq("1329227995784915872903807060280344576");
+
+            result = await this.forTest.decodeU128([1, "0x0001000000000000000000000000000000"]);
+            result.toString().should.eq("1329227995784915872903807060280344576");
+        });
+
+        it("should decodeI128 correctly", async () => {
+            result = await this.forTest.decodeI128([0, "0x00010000000000000000000000000000"]);
+            result.toString().should.eq("5192296858534827628530496329220096");
+
+            result = await this.forTest.decodeI128([1, "0x0000010000000000000000000000000000"]);
+            result.toString().should.eq("5192296858534827628530496329220096");
+        });
+
+        it("should decodeU256 correctly", async () => {
+            result = await this.forTest.decodeU256([0, "0x0100000000000000000000000000000000000000000000000000000000000000"]);
+            result.toString().should.eq("452312848583266388373324160190187140051835877600158453279131187530910662656");
+
+            result = await this.forTest.decodeU256([1, "0x000100000000000000000000000000000000000000000000000000000000000000"]);
+            result.toString().should.eq("452312848583266388373324160190187140051835877600158453279131187530910662656");
+        });
+
+        it("should decodeI256 correctly", async () => {
+            result = await this.forTest.decodeI256([0, "0x0001000000000000000000000000000000000000000000000000000000000000"]);
+            result.toString().should.eq("1766847064778384329583297500742918515827483896875618958121606201292619776");
+
+            result = await this.forTest.decodeI256([1, "0x000001000000000000000000000000000000000000000000000000000000000000"]);
+            result.toString().should.eq("1766847064778384329583297500742918515827483896875618958121606201292619776");
+        });
+
+        it("should decodeBool correctly", async () => {
+            result = await this.forTest.decodeBool([0, "0x00"]);
+            result.toString().should.eq("false");
+
+            result = await this.forTest.decodeBool([0, "0x01"]);
+            result.toString().should.eq("true");
+
+            result = await this.forTest.decodeBool([1, "0x0000"]);
+            result.toString().should.eq("false");
+
+            result = await this.forTest.decodeBool([1, "0x0001"]);
+            result.toString().should.eq("true");
+        });
+
+        it("should decodeBytes correctly", async () => {
+            result = await this.forTest.decodeBytes([0, "0x00000003425443"]);
+            result.toString().should.eq("0x425443");
+        });
+
+        it("should revert", async () => {
+            let offset = 100000;
+            await expectRevert(this.forTest.decodeU8(offset, "0x00"), 'invalid tuple value (coderType="tuple", value=100000)');
```

Comment:
It must be `[offset, "0x00"]` not `offset, "0x00"` and revert msg must be OBI: out of range